Adds E2E tests to the watchtower-plugin

sr-gi opened the pull request:
Adds some E2E tests for the CLN `watchtower-plugin`.

More may be nice, but I think this should suffice for now.

sr-gi commented:
Leaving this as a side note, in case it happens again:

I've seen tests failing in CI due to the plugin not being able to communicate with the tower (on register). I've tracked this down and the issue is related with the `http api` not being able to establish a gRPC connection with the `internal api`. This happens on the second test, so I'm wondering if it may be due to a wrong cleanup of `teosd` on the first one.

https://github.com/talaia-labs/rust-teos/runs/7000632548?check_suite_focus=true#step:9:424

meryacine commented on .github/workflows/cln-plugin.yaml:
```diff
@@ -0,0 +1,67 @@
+
+jobs:
+  cache-cln:
+    strategy:
```
nit: don't think this line nor the `fail-fast` line are needed since this is not a matrix build.
Same for the ones in the next job.

meryacine reviewed (commented):
Some early nits for the plugin's CI.

sr-gi commented on .github/workflows/cln-plugin.yaml:
```diff
@@ -0,0 +1,67 @@
+        run: |
+          sudo apt-get update && sudo apt-get install gettext
+          git clone https://github.com/ElementsProject/lightning.git && cd lightning && git checkout v0.11.0.1
+          pip install --user poetry && poetry install && source $(poetry env info -p)/bin/activate
```
It is, actually.

That's installing the Python libraries required to install CLN. In line 62 we are installing poetry again (give we are not catching the "binary") and using it to install the Python dependencies for the watchtower-plugin tests.

sr-gi commented on .github/workflows/cln-plugin.yaml:
```diff
@@ -0,0 +1,67 @@
+          git clone https://github.com/ElementsProject/lightning.git && cd lightning && git checkout v0.11.0.1
+          pip install --user poetry && poetry install && source $(poetry env info -p)/bin/activate
+          ./configure --enable-developer
+          make && sudo make install
```
It is. `pyln-testing` assumes `lightningd` and `lightning-cli` are installed. They are used to bootstrap networks during the tests (in our case a line network with 2 nodes).

sr-gi commented:
> Based on a search, looks like `OS 111 error` is fired when one trys to connect to some port which has no process listening on. I think maybe gRPC take some time to bootstrap and hook to the port while the http api makes its way in the async call faster, thus receiving a connection refused error. Just a theory.

That's my theory too. I've coded a pretty ugly patch to see if that was the case (https://github.com/talaia-labs/rust-teos/pull/65/commits/ec79e064dd9352339376ed38c7609cb178821ab1), but looks like reproducing is hard and CI is not hitting the issue now :S

sr-gi commented:
Looks like that's the issue indeed:

https://github.com/talaia-labs/rust-teos/runs/7027314645?check_suite_focus=true#step:9:4160

sr-gi commented:
@meryacine Addressed your comments and fixed the CI issue with the gRPC.

Got in touch with the `tonic` guys and looks like there is no way of signaling that the server has been completely bootstrapped, so a loop with a sleep for the client it .

meryacine commented:
> Got in touch with the `tonic` guys and looks like there is no way of signaling that the server has been completely bootstrapped, so a loop with a sleep for the client it is.

We can do the trigger thing though, like the shutdown signal. but would introduce more wires.

```rust
let (server_started_trigger, server_started_listener) = triggered::trigger();
let public_api_task = task::spawn(async move {
    Server::builder()
        .add_service(PublicTowerServicesServer::new(internal_rpc_api))
        .serve_with_shutdown(internal_rpc_api_addr, shutdown_signal_internal_rpc_api)
        .await
        .unwrap();
    server_started_trigger.trigger();
});
```

sr-gi commented:
@meryacine That's my bad actually. Instead of using my fork of the repo I created the CLN PRs pushing to upstream.

Not doing this anymore, so once the CLN PRs are merged we should only see CI on PRs not pushes (apart from master).
